cmd/minishift/cmd/oc_env.go

```diff
@@ -42,12 +45,19 @@ type OcShellConfig struct {
 }
 
 func getOcShellConfig(ocPath, forcedShell string) (*OcShellConfig, error) {
+	var cmdLine string
+
 	userShell, err := shell.GetShell(forcedShell)
 	if err != nil {
 		return nil, err
 	}
 
-	cmdLine := "minishift oc-env"
+	if cmdProfile.ProfileExists() {
+		cmdLine = "minishift oc-env --profile " + cmdProfile.GetProfileName()
```

Comment:
@LalatenduMohanty This relates to https://github.com/minishift/minishift/pull/1206#issuecomment-321843743. Can you explain what the plan is?